yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,139 @@
+            description: >
+                Parameters in required format to read the keyword data.
+      returns:
+          - name: keywordValue
+            type: variant[array[byte]]
+            description: >
+                On success, returns the Keyword's value read from the given
+                path. On failure, throws an exception.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - xyz.openbmc_project.Common.Error.InternalFailure
+
+    - name: GetLocationCode
+      description: >
+          Method to get location code from the D-Bus inventory path.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                D-Bus inventory path of the FRU.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                On success, returns location code. On failure, throws an
+                exception.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - xyz.openbmc_project.Common.Error.InternalFailure
+
+    - name: GetExpandedLocationCode
+      description: >
+          Method to get expanded location code for a given un-expanded location
+          code.
+      parameters:
+          - name: locationCode
+            type: string
+            description: >
+                Location code in un-expanded format.
+          - name: nodeNumber
+            type: uint16
+            description: >
+                Denotes the node in case of multi-node configuration. Defaulted
+                in case of single node configuration.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                Location code in expanded format.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - xyz.openbmc_project.Common.Error.InternalFailure
+
+    - name: deleteFRUVPD
+      description: >
+          Method to remove the FRU VPD.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                D-Bus inventory path of the FRU.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - xyz.openbmc_project.Common.Error.InternalFailure
+
+    - name: CollectFRUVPD
+      description: >
+          Method to perform single FRU VPD collection.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                D-Bus inventory path of the FRU.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - xyz.openbmc_project.Common.Error.InternalFailure
+
+properties:
+    - name: VPDCollectionStatus
+      type: enum[self.VPDStatus]
+      default: NotStarted
+      description: >
+          VPD collection status of the system.
+
+enumerations:
+    - name: VPDStatus
```

Comment:
just status or state will be good , do not add VPD everywhere as it is inside the VPD folder

Reply:
updated